src/Command/Email/ConfigurePlatformEmailCommand.php

```diff
@@ -0,0 +1,241 @@
+
+    $base_domain = $this->determineDomain($input);
+    $client = $this->cloudApiClientService->getClient();
+    $subscription = $this->determineCloudSubscription();
+    $response = $client->request('post', "/subscriptions/{$subscription->uuid}/domains", [
+      'form_params' => [
+        'domain' => $base_domain,
+      ],
+    ]);
+    // @todo Check response!
+
+    $domain_uuid = $this->fetchDomainUuid($client, $subscription, $base_domain);
+    $this->createDnsText($client, $subscription, $domain_uuid);
+
+    $this->io->success([
+      "Great! You've registered the domain {$base_domain} to subscription {$subscription->name}.",
+      "We created dns-records.txt",
+      "Provide these records to your DNS provider",
+      "After you've done this, continue."
+    ]);
+    $continue = $this->io->confirm('Have you finished providing the DNS records to your DNS provider?');
+    if (!$continue) {
+      return 1;
+    }
+
+    $this->pollDomainRegistrationsUntilSuccess($subscription, $domain_uuid, $this->output);
+    $this->io->success("The next step is associating your verified domain with an application (or applications) in the subscription where your domain has been registered.");
+
+    $this->addDomainToSubscriptionApplications($client, $subscription, $base_domain, $domain_uuid);
+    $this->io->success("You're all set to start using Platform Email!");
+
+    return 0;
+  }
+
+  /**
+   * @param Client $client
+   * @param SubscriptionResponse $subscription
+   * @param string $base_domain
+   * @param string $domain_uuid
+   *
+   * @return int|void
+   */
+  protected function addDomainToSubscriptionApplications(Client $client, SubscriptionResponse $subscription, string $base_domain, string $domain_uuid) {
+    $applications_resource = new Applications($client);
+    $applications = $applications_resource->getAll();
+    $subscription_applications = [];
+    foreach ($applications as $application) {
+      if ($application->subscription->uuid === $subscription->uuid) {
+        $subscription_applications[] = $application;
+      }
+    }
+    if (count($subscription_applications) === 0) {
+      $this->io->error("You do not have access to any applications on the {$subscription->name} subscription");
+      return 1;
+    }
+    elseif (count($subscription_applications) === 1) {
+      $applications = $subscription_applications;
+    }
+    else {
+      $applications = $this->promptChooseFromObjectsOrArrays($subscription_applications, 'uuid', 'name', "What are the applications you'd like to associate this domain with? You may enter multiple separated by a comma.", TRUE);
+    }
+
+    $environments_resource = new Environments($client);
+    foreach ($applications as $application) {
+      $response = $client->request('post', "/applications/{$application->uuid}/email/domains/{$domain_uuid}/actions/associate");
+      // @todo Check response!
```

Comment:
Ditto.

Reply:
To @grasmash 's comment above about how we can skip the response check if this throws an exception, this request throws an exception as well.

<img width="926" alt="Screen Shot 2022-02-17 at 1 03 39 PM" src="https://user-images.githubusercontent.com/28101298/154543188-93b46437-3c47-4aa3-b19e-4dda4b1c3a86.png">

Reply:
Great.